cpp/ql/src/experimental/Best Practices/UselessTest.ql

```diff
@@ -0,0 +1,43 @@
+/**
+ * @name Useless Test
+ * @description A boolean condition that is guaranteed to never be evaluated should be deleted.
+ * @kind problem
+ * @problem.severity warning
+ * @id cpp/uselesstest
+ * @tags reliability
+ *       readability
+ */
+
+import cpp
+import semmle.code.cpp.valuenumbering.GlobalValueNumbering
+
+predicate sameExpr(Expr e1, Expr e2) { globalValueNumber(e1).getAnExpr() = e2 }
+
+Element nearestParent(Expr e) {
+  if
+    e.getParent().(Expr).getFullyConverted() instanceof ParenthesisExpr or
```

Comment:
I think you want `getConverted*()` rather than `getFullyConverted` - if there's an explicit cast of a parenthesized expression, like `(short)(a + b)`, `getFullyConverted()` will be the cast rather than the parenthesis.